patcher.sh

```diff
@@ -0,0 +1,302 @@
+RESOURCES="${OUTPUT}/assets"
+
+patch_am2r ()
+{
+	# Set prefix to default value if empty
+	if [ -z "$PREFIX" ]; then
+		if [ "$SYSTEMWIDE" = true ]; then
+			PREFIX="/usr/local"
+		else
+			PREFIX="$SCRIPT_DIR"
+		fi
+	fi
+
+	# We need to do variable adjustments based on the prefix
+	if [ "$SYSTEMWIDE" = true ]; then
+		OUTPUT="${PREFIX}/opt/am2r"
+		RESOURCES="${OUTPUT}/assets"
+	else
+		OUTPUT="${PREFIX}/am2r_${VERSION}"
+		RESOURCES="${OUTPUT}/assets"
+	fi
+	# Cleanup in case the dirs/files exists
+	cleanupDirs=("$OUTPUT" "$SCRIPT_DIR/AM2R.AppDir" "$SCRIPT_DIR/AM2R-x86_64.AppImage" "$SCRIPT_DIR/AM2RWrapper/" "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper.apk")
+	rm -rf "${cleanupDirs[@]}"
+
+	mkdir -p "$OUTPUT"
+	mkdir -p "$RESOURCES"
+
+	# Check for AM2R_11
+	if [[ -f "$AM2RZIP" ]]; then
+		echo "AM2R_11.zip found!"
+		unzip -q "$AM2RZIP" -d "$OUTPUT"
+	elif [[ -d "$AM2RZIP" ]]; then
+		echo "AM2R_11 folder found!"
+		cp -R "$AM2RZIP" "$OUTPUT"
+	else
+		echo "AM2R_11 not found! Please place AM2R_11.zip (case sensitive) into \"$SCRIPT_DIR\" or provide it via command line arguments and try again."
+		return 1
+	fi
+
+	# Check for which OS we patch
+	if [ "$OSCHOICE" = "linux" ]; then
+
+		echo "Patching for Linux..."
+		echo "Applying AM2R xdelta patch..."
+		xdelta3 -dfs "$OUTPUT/AM2R.exe" "$SCRIPT_DIR/data/AM2R.xdelta" "$OUTPUT/runner"
+
+		echo "Applying data xdelta patch..."
+		xdelta3 -dfs "$OUTPUT/data.win" "$SCRIPT_DIR/data/game.xdelta" "$RESOURCES/game.unx"
+
+		echo "Cleaning up residual AM2R 1.1 files..."
+		rm "$OUTPUT/AM2R.exe" "$OUTPUT/data.win" "$OUTPUT/D3DX9_43.dll"
+
+		echo "Formatting game directory..."
+		# This won't move the runner, because doesn't have a dot
+		mv $OUTPUT/*.* "$RESOURCES/"
+
+		echo "Installing new datafiles..."
+		cp -R "$SCRIPT_DIR"/data/files_to_copy/* "$RESOURCES/"
+
+		if [ "$HQMUSIC" = true ]; then
+			echo "Copying HQ music..."
+			cp "$SCRIPT_DIR"/data/HDR_HQ_in-game_music/*.ogg "$RESOURCES/"
+		fi
+
+		# Format music for Unix, aka lowercase it
+		# Finds every file, and if its not already lowercase, lowercases it.
+		find "$RESOURCES" -type f -exec \
+			bash -c 'target="{}";
+				cd "$(dirname "${target}")";
+				target="$(basename "${target}")";
+				! [[ "${target}" = "${target,,}" ]] && mv "${target}" "${target,,}"' \;
+
+		# Remove old lang folder
+		rm -R "$OUTPUT"/lang
+
+		chmod +x "$OUTPUT/runner"
+
+		if [ "$SYSTEMWIDE" = true ]; then
+			mkdir "$PREFIX/bin"
+			ln -s "$OUTPUT/runner" "$PREFIX/bin/am2r"
+		fi
+
+		# Create .desktop file
+		echo "Creating desktop file..."
+		cp "$SCRIPT_DIR/data/files_to_copy/icon.png" "$OUTPUT/icon.png"
+		desktopPath="$OUTPUT/AM2R.desktop"
+		if [ "$SYSTEMWIDE" = true ]; then
+			mkdir -p "$PREFIX/share/applications"
+			desktopPath="$PREFIX/share/applications/AM2R.desktop"
+		fi
+		cp "$SCRIPT_DIR/DesktopTemplate" "$desktopPath"
+
+		# replace [REPLACE] with OUTPUT.Replace("/", "\/") in desktop file
```

Comment:
Any reason the template doesn't have this in the first place?

Reply:
Because it's not possible to know what "OUTPUT" is before running the script.
Comment is probably misleading a bit, as its more c#-code, but basically what it's supposed to say, is that all `/` in `OUTPUT` get replaced with `\/`, because otherwise there were some kind of issues with running the desktop file that I can't remember right now.

Reply:
> is that all `/` in OUTPUT get replaced with `\/`

That might be because you're using sed, which needs to have backslashes escaped. Desktop files themselves should not need that.

Reply:
Ah yeah, was a sed thing my bad. Was because $OUTPUT contains a file path (i.e. `./foo/bar`), but when expanded to the full sed command, it'd be `sed -i "s/\[REPLACE\]/./foo/bar/"` which sed interpreted as "replace '[REPLACE]' with '.'".